spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/integration/inbound/PubSubInboundChannelAdapter.java

```diff
@@ -0,0 +1,53 @@
+package org.springframework.cloud.gcp.pubsub.integration.inbound;
+
+import com.google.cloud.pubsub.spi.v1.AckReplyConsumer;
+import com.google.cloud.pubsub.spi.v1.Subscriber;
+import com.google.pubsub.v1.PubsubMessage;
+import com.google.pubsub.v1.SubscriptionName;
+import java.util.HashMap;
+import java.util.Map;
+import org.springframework.integration.endpoint.MessageProducerSupport;
+import org.springframework.messaging.MessageHeaders;
+
+/**
+ * Created by joaomartins on 5/19/17.
+ */
+public class PubSubInboundChannelAdapter extends MessageProducerSupport {
+
+  private String projectId;
+  private String subscriptionName;
+  private Subscriber subscriber;
+
+  public PubSubInboundChannelAdapter(String projectId, String subscriptionName) {
+    this.projectId = projectId;
+    this.subscriptionName = subscriptionName;
+  }
+
+  @Override
+  protected void onInit() {
+    super.onInit();
+
+    subscriber = Subscriber.defaultBuilder(SubscriptionName.create(
+        projectId, this.subscriptionName), this::receiveMessage).build();
+    subscriber.startAsync();
+  }
+
+  private void receiveMessage(PubsubMessage message, AckReplyConsumer consumer) {
+    Map<String, Object> messageHeaders = new HashMap<>();
+    message.getAttributesMap().forEach(messageHeaders::put);
+
+    sendMessage(getMessagingTemplate().getMessageConverter().toMessage(
+        message.getData(),
+        new MessageHeaders(messageHeaders)));
+    consumer.ack();
```

Comment:
The `AckReplyConsumer` should go to message headers.
I can decide downstream in the flow when and how to `ack()` it. 
See similar logic in the `KinesisMessageDrivenChannelAdapter`: https://github.com/spring-projects/spring-integration-aws/blob/master/src/main/java/org/springframework/integration/aws/inbound/kinesis/KinesisMessageDrivenChannelAdapter.java#L852
or `KafkaMessageDrivenChannelAdapter`: https://github.com/spring-projects/spring-integration-kafka/blob/master/src/main/java/org/springframework/integration/kafka/inbound/KafkaMessageDrivenChannelAdapter.java#L304